### PR TITLE
Add ZIO2 context propagation

### PR DESCRIPTION
**Pull request by dispalt**
I copied the CE3 tests, but made them work in ZIO style.

### CODE REVIEW

File: instrumentation/kamon-zio-2/src/main/scala/kamon/instrumentation/zio2/ZIO2Instrumentation.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  override def onSuspend[E, A_](fiber: Fiber.Runtime[E, A_])(implicit unsafe: Unsafe): Unit = {
+    val pctx = Kamon.currentContext()
+    fiber.asInstanceOf[HasContext].setContext(pctx)
+  }
+
+  override def onResume[E, A_](fiber: Fiber.Runtime[E, A_])(implicit unsafe: Unsafe): Unit = {
+    val ctx = fiber.asInstanceOf[HasContext].context
+    Kamon.storeContext(ctx)
+  }
```
Comment (m50d) [inline instrumentation/kamon-zio-2/src/main/scala/kamon/instrumentation/zio2/ZIO2Instrumentation.scala:57]: Are these the right way around? That looks opposite from what I'd expect but IDK.

**Review by m50d — APPROVED** [review]

File: instrumentation/kamon-zio-2/src/main/scala/kamon/instrumentation/zio2/ZIO2Instrumentation.scala (hunk 1 above)
Comment (dispalt) [reply instrumentation/kamon-zio-2/src/main/scala/kamon/instrumentation/zio2/ZIO2Instrumentation.scala]: on suspend you're setting the context of the fiber that you had back to the fiber itself where you were executing it. 

onResume your taking the fiber from the pool that you are resuming and applying context to the current session.  I am probably saying that wrong.

**Comment by dispalt** [comment]
@ivantopo would love for you to take a look

**Review by hughsimpson — APPROVED** [review]

**Comment by hughsimpson** [comment]
lgtm

**Pull request merged**